### ISSUE TITLE
Not showing

### ISSUE THREAD

**Issue by khwerhahn** [tj/node-progress#104]
Does the terminal need to have any special settings? Using zsh... 

Nothing happens when I use the example in the repo.

**Comment by hallas** [comment]
Nope nothing special, very weird, can you tell us anything more?

**Comment by khwerhahn** [comment]
Not really... Followed the example and the ``` if (bar.complete) ``` part works, but the progress bar doesn't show up

```
var ProgressBar = require('progress');

var bar = new ProgressBar(':bar', { total: 10 });
var timer = setInterval(function () {
  bar.tick();
  if (bar.complete) {
    console.log('\ncomplete\n');
    clearInterval(timer);
  }
}, 100);
```

**Comment by hallas** [comment]
I just tested this is zsh on Mac OS X: zsh 5.0.5 (x86_64-apple-darwin14.0)

What does your "zsh --version" give you?

**Comment by khwerhahn** [comment]
Same version.

**Comment by hallas** [comment]
Well maybe some kind of setting you have on then?

**Comment by wzshuang** [comment]
the progressbar shows in cmd window, but doesn't show in webstorm ide's console on windows, any thing wrong?

**Comment by nullivex** [comment]
@wzshuang as far as I know it has to do with the way it draws and webstorms terminal emulation which is not displaying a raw terminal. I have similar results with testing suites etc.

I dont believe it is something that we can affect from within the code.

**Comment by wzshuang** [comment]
@nullivex I debuged the code and note that 'this.stream.isTTY' is undefined, I don't know what TTY is, I will google it.

**Comment by hallas** [comment]
@wzshuang did you change the context of this?

**Comment by wzshuang** [comment]
@hallas no, I didn't change the source code, just debug it.

**Comment by ishmangit** [comment]
I'm having the same issue on Linux (Elementary OS -Freya-)

**Comment by slavafomin** [comment]
Just stumbled upon this problem too. Trying to use it in WebStorm IDE, but it's not rendering. Are there any news on this? It would be very convenient to be able to see the progress inside the IDE window.

**Comment by wanjas** [comment]
A long time has passed since but... 
In short, the issue with WebStrom is that console there isn't TTY that is required for _progress_ to work. 

There is a workaround for that (not without some tradeoffs). Take a look at the latest comment from Sergey Simonchik on JetBrains tracker.

[https://youtrack.jetbrains.com/issue/WEB-1933](https://youtrack.jetbrains.com/issue/WEB-1933#focus=streamItem-27-3141105.0-0)

Quote: 

> Fixed in WebStorm 2018.2 if nodejs.console.use.terminal registry key is enabled - for that, please do the following:
> * "Help | Find Action..." on the main menu;
> * Type "registry" and click "Registry..." found element;
> * Find _**nodejs.console.use.terminal**_ key and enable it.

**Comment by zjr** [comment]
Thank you for finding that @wanjas.

**Comment by PQMISSCMP** [comment]
Same problem...in local shell work well but on the cloud, when i look console i don't view the progress bar
